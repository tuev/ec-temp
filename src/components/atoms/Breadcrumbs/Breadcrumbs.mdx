# Stories mdx for Breadcrumbs

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Breadcrumbs.stories.tsx'
import Breadcrumbs from './Breadcrumbs.tsx'

<Story name="basic" />