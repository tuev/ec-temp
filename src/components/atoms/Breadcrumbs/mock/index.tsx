import React, { FC } from 'react'
import Typography from '@material-ui/core/Typography'
import Link from '@material-ui/core/Link'

const items: FC = () => (
  <>
    <Link color="inherit" href="/">
      Home
    </Link>
    <Link color="inherit" href="/getting-started/installation/">
      All Categories
    </Link>
    <Typography color="textPrimary">
      Men&apos;s Clothing & Accessories
    </Typography>
  </>
)

export default items
