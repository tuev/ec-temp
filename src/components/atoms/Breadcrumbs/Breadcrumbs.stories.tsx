/**
 *
 *  Stories for Breadcrumbs
 *
 */

import React from 'react'
import Breadcrumbs from './Breadcrumbs'

import Items from './mock'

export default {
  component: Breadcrumbs,
  includeStories: ['basic'],
  title: 'Breadcrumbs',
}

export const basic = () => (
  <Breadcrumbs>
    <Items />
  </Breadcrumbs>
)
