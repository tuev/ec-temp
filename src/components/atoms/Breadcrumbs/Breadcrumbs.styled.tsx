/**
 *
 * Styled for Breadcrumbs
 *
 */

import styled from 'styled-components'
import Breadcrumbs from '@material-ui/core/Breadcrumbs'
import { BaseBreadcrumbsProps } from './Breadcrumbs.types'

export const AppBreadcrumbs: React.FC<BaseBreadcrumbsProps> = styled(
  Breadcrumbs
)<BaseBreadcrumbsProps>``
