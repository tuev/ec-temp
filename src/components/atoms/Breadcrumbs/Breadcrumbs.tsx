/**
 *
 * Breadcrumbs
 *
 */

import React, { FC } from 'react'
import { BaseBreadcrumbsProps } from './Breadcrumbs.types'
import { AppBreadcrumbs } from './Breadcrumbs.styled'

const Breadcrumbs: FC<BaseBreadcrumbsProps> = (props: BaseBreadcrumbsProps) => {
  return (
    <AppBreadcrumbs separator="â€¢" aria-label="breadcrumb">
      {props.children}
    </AppBreadcrumbs>
  )
}

export default Breadcrumbs
