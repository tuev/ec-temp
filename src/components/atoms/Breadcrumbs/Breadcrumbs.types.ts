import { SpaceProps } from 'styled-system'
import { BreadcrumbsProps } from '@material-ui/core/Breadcrumbs'

export declare type BaseBreadcrumbsProps = BreadcrumbsProps & SpaceProps
