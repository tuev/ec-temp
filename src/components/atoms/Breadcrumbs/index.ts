/**
 *
 * Index for Breadcrumbs
 *
 */

import Breadcrumbs from './Breadcrumbs'

export default Breadcrumbs
