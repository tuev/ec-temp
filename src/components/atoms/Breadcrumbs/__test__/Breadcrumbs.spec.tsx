/**
 *
 * Tests for Breadcrumbs
 *
 */

import React from 'react'
import { mount } from 'enzyme'

import Breadcrumbs from '..'
import Items from '../mock'

describe('App Breadcrumbs', () => {
  it('Test get default Breadcrumbs no params ', () => {
    // test snapshot
    const wrapper = mount(
      <Breadcrumbs>
        <Items />
      </Breadcrumbs>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
