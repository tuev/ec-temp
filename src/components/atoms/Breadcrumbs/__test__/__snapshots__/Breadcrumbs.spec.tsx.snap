// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Breadcrumbs Test get default Breadcrumbs no params  1`] = `
<Breadcrumbs>
  <Styled(WithStyles(ForwardRef(Breadcrumbs)))
    aria-label="breadcrumb"
    separator="•"
  >
    <WithStyles(ForwardRef(Breadcrumbs))
      aria-label="breadcrumb"
      className=""
      separator="•"
    >
      <ForwardRef(Breadcrumbs)
        aria-label="breadcrumb"
        className=""
        classes={
          Object {
            "li": "MuiBreadcrumbs-li",
            "ol": "MuiBreadcrumbs-ol",
            "root": "MuiBreadcrumbs-root",
            "separator": "MuiBreadcrumbs-separator",
          }
        }
        separator="•"
      >
        <WithStyles(ForwardRef(Typography))
          aria-label="breadcrumb"
          className="MuiBreadcrumbs-root "
          color="textSecondary"
          component="nav"
        >
          <ForwardRef(Typography)
            aria-label="breadcrumb"
            className="MuiBreadcrumbs-root "
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
            color="textSecondary"
            component="nav"
          >
            <nav
              aria-label="breadcrumb"
              className="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              <ol
                className="MuiBreadcrumbs-ol"
              >
                <li
                  className="MuiBreadcrumbs-li"
                  key="child-0"
                >
                  <items
                    key=".0"
                  >
                    <WithStyles(ForwardRef(Link))
                      color="inherit"
                      href="/"
                    >
                      <ForwardRef(Link)
                        classes={
                          Object {
                            "button": "MuiLink-button",
                            "focusVisible": "Mui-focusVisible",
                            "root": "MuiLink-root",
                            "underlineAlways": "MuiLink-underlineAlways",
                            "underlineHover": "MuiLink-underlineHover",
                            "underlineNone": "MuiLink-underlineNone",
                          }
                        }
                        color="inherit"
                        href="/"
                      >
                        <WithStyles(ForwardRef(Typography))
                          className="MuiLink-root MuiLink-underlineHover"
                          color="inherit"
                          component="a"
                          href="/"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          variant="inherit"
                        >
                          <ForwardRef(Typography)
                            className="MuiLink-root MuiLink-underlineHover"
                            classes={
                              Object {
                                "alignCenter": "MuiTypography-alignCenter",
                                "alignJustify": "MuiTypography-alignJustify",
                                "alignLeft": "MuiTypography-alignLeft",
                                "alignRight": "MuiTypography-alignRight",
                                "body1": "MuiTypography-body1",
                                "body2": "MuiTypography-body2",
                                "button": "MuiTypography-button",
                                "caption": "MuiTypography-caption",
                                "colorError": "MuiTypography-colorError",
                                "colorInherit": "MuiTypography-colorInherit",
                                "colorPrimary": "MuiTypography-colorPrimary",
                                "colorSecondary": "MuiTypography-colorSecondary",
                                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                "displayBlock": "MuiTypography-displayBlock",
                                "displayInline": "MuiTypography-displayInline",
                                "gutterBottom": "MuiTypography-gutterBottom",
                                "h1": "MuiTypography-h1",
                                "h2": "MuiTypography-h2",
                                "h3": "MuiTypography-h3",
                                "h4": "MuiTypography-h4",
                                "h5": "MuiTypography-h5",
                                "h6": "MuiTypography-h6",
                                "noWrap": "MuiTypography-noWrap",
                                "overline": "MuiTypography-overline",
                                "paragraph": "MuiTypography-paragraph",
                                "root": "MuiTypography-root",
                                "srOnly": "MuiTypography-srOnly",
                                "subtitle1": "MuiTypography-subtitle1",
                                "subtitle2": "MuiTypography-subtitle2",
                              }
                            }
                            color="inherit"
                            component="a"
                            href="/"
                            onBlur={[Function]}
                            onFocus={[Function]}
                            variant="inherit"
                          >
                            <a
                              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                              href="/"
                              onBlur={[Function]}
                              onFocus={[Function]}
                            >
                              Home
                            </a>
                          </ForwardRef(Typography)>
                        </WithStyles(ForwardRef(Typography))>
                      </ForwardRef(Link)>
                    </WithStyles(ForwardRef(Link))>
                    <WithStyles(ForwardRef(Link))
                      color="inherit"
                      href="/getting-started/installation/"
                    >
                      <ForwardRef(Link)
                        classes={
                          Object {
                            "button": "MuiLink-button",
                            "focusVisible": "Mui-focusVisible",
                            "root": "MuiLink-root",
                            "underlineAlways": "MuiLink-underlineAlways",
                            "underlineHover": "MuiLink-underlineHover",
                            "underlineNone": "MuiLink-underlineNone",
                          }
                        }
                        color="inherit"
                        href="/getting-started/installation/"
                      >
                        <WithStyles(ForwardRef(Typography))
                          className="MuiLink-root MuiLink-underlineHover"
                          color="inherit"
                          component="a"
                          href="/getting-started/installation/"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          variant="inherit"
                        >
                          <ForwardRef(Typography)
                            className="MuiLink-root MuiLink-underlineHover"
                            classes={
                              Object {
                                "alignCenter": "MuiTypography-alignCenter",
                                "alignJustify": "MuiTypography-alignJustify",
                                "alignLeft": "MuiTypography-alignLeft",
                                "alignRight": "MuiTypography-alignRight",
                                "body1": "MuiTypography-body1",
                                "body2": "MuiTypography-body2",
                                "button": "MuiTypography-button",
                                "caption": "MuiTypography-caption",
                                "colorError": "MuiTypography-colorError",
                                "colorInherit": "MuiTypography-colorInherit",
                                "colorPrimary": "MuiTypography-colorPrimary",
                                "colorSecondary": "MuiTypography-colorSecondary",
                                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                "displayBlock": "MuiTypography-displayBlock",
                                "displayInline": "MuiTypography-displayInline",
                                "gutterBottom": "MuiTypography-gutterBottom",
                                "h1": "MuiTypography-h1",
                                "h2": "MuiTypography-h2",
                                "h3": "MuiTypography-h3",
                                "h4": "MuiTypography-h4",
                                "h5": "MuiTypography-h5",
                                "h6": "MuiTypography-h6",
                                "noWrap": "MuiTypography-noWrap",
                                "overline": "MuiTypography-overline",
                                "paragraph": "MuiTypography-paragraph",
                                "root": "MuiTypography-root",
                                "srOnly": "MuiTypography-srOnly",
                                "subtitle1": "MuiTypography-subtitle1",
                                "subtitle2": "MuiTypography-subtitle2",
                              }
                            }
                            color="inherit"
                            component="a"
                            href="/getting-started/installation/"
                            onBlur={[Function]}
                            onFocus={[Function]}
                            variant="inherit"
                          >
                            <a
                              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                              href="/getting-started/installation/"
                              onBlur={[Function]}
                              onFocus={[Function]}
                            >
                              All Categories
                            </a>
                          </ForwardRef(Typography)>
                        </WithStyles(ForwardRef(Typography))>
                      </ForwardRef(Link)>
                    </WithStyles(ForwardRef(Link))>
                    <WithStyles(ForwardRef(Typography))
                      color="textPrimary"
                    >
                      <ForwardRef(Typography)
                        classes={
                          Object {
                            "alignCenter": "MuiTypography-alignCenter",
                            "alignJustify": "MuiTypography-alignJustify",
                            "alignLeft": "MuiTypography-alignLeft",
                            "alignRight": "MuiTypography-alignRight",
                            "body1": "MuiTypography-body1",
                            "body2": "MuiTypography-body2",
                            "button": "MuiTypography-button",
                            "caption": "MuiTypography-caption",
                            "colorError": "MuiTypography-colorError",
                            "colorInherit": "MuiTypography-colorInherit",
                            "colorPrimary": "MuiTypography-colorPrimary",
                            "colorSecondary": "MuiTypography-colorSecondary",
                            "colorTextPrimary": "MuiTypography-colorTextPrimary",
                            "colorTextSecondary": "MuiTypography-colorTextSecondary",
                            "displayBlock": "MuiTypography-displayBlock",
                            "displayInline": "MuiTypography-displayInline",
                            "gutterBottom": "MuiTypography-gutterBottom",
                            "h1": "MuiTypography-h1",
                            "h2": "MuiTypography-h2",
                            "h3": "MuiTypography-h3",
                            "h4": "MuiTypography-h4",
                            "h5": "MuiTypography-h5",
                            "h6": "MuiTypography-h6",
                            "noWrap": "MuiTypography-noWrap",
                            "overline": "MuiTypography-overline",
                            "paragraph": "MuiTypography-paragraph",
                            "root": "MuiTypography-root",
                            "srOnly": "MuiTypography-srOnly",
                            "subtitle1": "MuiTypography-subtitle1",
                            "subtitle2": "MuiTypography-subtitle2",
                          }
                        }
                        color="textPrimary"
                      >
                        <p
                          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
                        >
                          Men's Clothing & Accessories
                        </p>
                      </ForwardRef(Typography)>
                    </WithStyles(ForwardRef(Typography))>
                  </items>
                </li>
              </ol>
            </nav>
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
      </ForwardRef(Breadcrumbs)>
    </WithStyles(ForwardRef(Breadcrumbs))>
  </Styled(WithStyles(ForwardRef(Breadcrumbs)))>
</Breadcrumbs>
`;
